testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -213,6 +213,57 @@ public void testSparkParquetTimestampCompatibility(String sparkTimestampFormat,
         onSpark().executeQuery("DROP TABLE " + sparkTableName);
     }
 
+    @Test(groups = {HIVE_SPARK, PROFILE_SPECIFIC_TESTS})
+    public void testInsertFailsInBucketedTableWithoutBucketingVersionTableParameter()
+    {
+        String hiveTableName = "spark_insert_bucketed_table_" + randomTableSuffix();
+
+        onSpark().executeQuery(
+                "CREATE TABLE default." + hiveTableName + "(a_key integer, a_value integer) " +
+                        "USING PARQUET " +
+                        "TBLPROPERTIES ('transactional'='false') " +
+                        "CLUSTERED BY (a_key) INTO 3 BUCKETS");
+
+        assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO default." + hiveTableName + " VALUES (1, 100)"))
+                .hasMessageContaining(format("Inserting into Spark bucketed table default.%s is not supported", hiveTableName));
+
+        onSpark().executeQuery("DROP TABLE " + hiveTableName);
+    }
+
+    @Test(groups = {HIVE_SPARK, PROFILE_SPECIFIC_TESTS})
+    public void testUpdateFailsInBucketedTableWithoutBucketingVersionTableParameter()
+    {
+        String hiveTableName = "spark_update_bucketed_table_" + randomTableSuffix();
+
+        onSpark().executeQuery(
+                "CREATE TABLE default." + hiveTableName + "(a_key integer, a_value integer) " +
+                        "USING ORC " +
+                        "TBLPROPERTIES ('transactional'='true') " +
```

Comment:
We can test - but it feels weird. Can such a table realistically be created in Spark? Does spark support ORC ACID tables format. Or it would still write data as it was non-transactional table?

Reply:
Note that the spark bucketed check happens after the check

```
if (!isFullAcidTable(table.getParameters())) {
            throw new TrinoException(NOT_SUPPORTED, "Hive update is only supported for ACID transactional tables");
        }
```

so I think that Spark does create ORC ACID tables.

Reply:
But doesn't it come from the fact that you explicitly put `TBLPROPERTIES ('transactional'='true')` in `CREATE TABLE` statement. 
Would that make sense for Spark users to put sht like that? (I guess Spark will pass all the options verbatim, but not necessarily make any use of them).

Reply:
If anything can depend on this, we should test with both: `TBLPROPERTIES ('transactional'='true')` set in Spark, and not set. (maybe skipping testing with `TBLPROPERTIES ('transactional'='false')`...)

Reply:
```
+----------------------------------------------------+
|                   createtab_stmt                   |
+----------------------------------------------------+
| CREATE TABLE `spark_update_bucketed_table_43s315y438ky`( |
|   `a_key` int,                                     |
|   `a_value` int)                                   |
| ROW FORMAT SERDE                                   |
|   'org.apache.hadoop.hive.ql.io.orc.OrcSerde'      |
| WITH SERDEPROPERTIES (                             |
|   'path'='hdfs://hadoop-master:9000/user/hive/warehouse/spark_update_bucketed_table_43s315y438ky')  |
| STORED AS INPUTFORMAT                              |
|   'org.apache.hadoop.hive.ql.io.orc.OrcInputFormat'  |
| OUTPUTFORMAT                                       |
|   'org.apache.hadoop.hive.ql.io.orc.OrcOutputFormat' |
| LOCATION                                           |
|   'hdfs://hadoop-master:9000/user/hive/warehouse/spark_update_bucketed_table_43s315y438ky' |
| TBLPROPERTIES (                                    |
|   'spark.sql.create.version'='3.1.2',              |
|   'spark.sql.sources.provider'='ORC',              |
|   'spark.sql.sources.schema.bucketCol.0'='a_key',  |
|   'spark.sql.sources.schema.numBucketCols'='1',    |
|   'spark.sql.sources.schema.numBuckets'='3',       |
|   'spark.sql.sources.schema.numParts'='1',         |
|   'spark.sql.sources.schema.part.0'='{"type":"struct","fields":[{"name":"a_key","type":"integer","nullable":true,"metadata":{}},{"name":"a_value","type":"integer","nullable":true,"metadata":{}}]}',  |
|   'transactional'='true',                          |
|   'transient_lastDdlTime'='1643892072')            |
+----------------------------------------------------+
```

I had a look in Hive and Spark does indeed create the table as transactional even when the table parameter `TBLPROPERTIES ('transactional'='true')` is missing.